internal/services/redhatopenshift/redhat_openshift_cluster_resource_test.go

```diff
@@ -565,6 +580,103 @@ resource "azurerm_redhat_openshift_cluster" "test" {
   `, r.template(data), data.RandomInteger, data.RandomString)
 }
 
+func (r OpenShiftClusterResource) preconfiguredNSG(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_network_security_group" "test" {
+  name                = "test-nsg"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_network_security_rule" "test_allow_all_inbound" {
+  name                        = "test_allow_all_inbound"
+  resource_group_name         = azurerm_resource_group.test.name
+  priority                    = 100
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "*"
+  source_port_range           = "*"
+  destination_port_range      = "*"
+  source_address_prefix       = "*"
+  destination_address_prefix  = "*"
+  network_security_group_name = azurerm_network_security_group.test.name
+}
+
+resource "azurerm_network_security_rule" "test_allow_all_outbound" {
+  name                        = "test_allow_all_outbound"
+  resource_group_name         = azurerm_resource_group.test.name
+  priority                    = 100
+  direction                   = "Outbound"
+  access                      = "Allow"
+  protocol                    = "*"
+  source_port_range           = "*"
+  destination_port_range      = "*"
+  source_address_prefix       = "*"
+  destination_address_prefix  = "*"
+  network_security_group_name = azurerm_network_security_group.test.name
+}
+
+resource "azurerm_subnet_network_security_group_association" "test_main" {
+  subnet_id                 = azurerm_subnet.main_subnet.id
+  network_security_group_id = azurerm_network_security_group.test.id
+}
+
+resource "azurerm_subnet_network_security_group_association" "test_worker" {
+  subnet_id                 = azurerm_subnet.worker_subnet.id
+  network_security_group_id = azurerm_network_security_group.test.id
+}
+
+resource "azurerm_redhat_openshift_cluster" "test" {
+  name                = "acctestaro%[2]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  cluster_profile {
+    domain       = "aro-%[3]s.com"
+    version      = "4.13.23"
+  }
+
+  network_profile {
+    pod_cidr                  = "10.128.0.0/14"
+    service_cidr              = "172.30.0.0/16"
+    preconfigured_nsg_enabled = true
```

Comment:
(as above)

```suggestion
    preconfigured_network_security_group_enabled = true
```

Reply:
Apologies, I missed the fact that you made this change and I simply needed to commit.  Regardless, I fixed in commit becf28602323ab08a37def639367eb0975a74629